qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -595,56 +523,36 @@ def refresh(self) -> None:
             IBMJobApiError: If an unexpected error occurred when communicating
                 with the server.
         """
+        # TODO: Change to use runtime response data as much as possible
         with api_to_job_error():
-            api_response = self._api_client.job_get(self.job_id())
+            api_response = self._runtime_client.job_get(self.job_id())
+            api_metadata = self._runtime_client.job_metadata(self.job_id())
 
         try:
-            api_response.pop("job_id")
-            self._creation_date = dateutil.parser.isoparse(
-                api_response.pop("creation_date")
-            )
-            self._api_status = api_response.pop("status")
-            if "kind" in api_response:
-                self._kind = ApiJobKind(api_response.pop("kind"))
-            if "qobj" in api_response:
-                self._qobj = dict_to_qobj(api_response.pop("qobj"))
+            api_response.pop("id")
+            self._creation_date = dateutil.parser.isoparse(api_response.pop("created"))
+            self._api_status = api_response.pop("state")["status"]
         except (KeyError, TypeError) as err:
             raise IBMJobApiError(
                 "Unexpected return value received " "from the server: {}".format(err)
             ) from err
 
-        self._name = api_response.pop("name", None)
         self._time_per_step = api_response.pop("time_per_step", None)
-        self._error = api_response.pop("error", None)
         self._tags = api_response.pop("tags", [])
-        self._run_mode = api_response.pop("run_mode", None)
-        self._use_object_storage = self._kind == ApiJobKind.QOBJECT_STORAGE
         self._status, self._queue_info = self._get_status_position(
-            self._api_status, api_response.pop("info_queue", None)
+            self._api_status, api_metadata
         )
         self._client_version = self._extract_client_version(
-            api_response.pop("client_info", None)
+            api_metadata.get("qiskit_version", None)
         )
-        self._set_result(api_response.pop("result", None))
+        if self._status == JobStatus.DONE:
+            api_result = self._runtime_client.job_results(self.job_id())
+            self._set_result(api_result)
 
         for key, value in api_response.items():
             self._data[key + "_"] = value
         self._refreshed = True
 
-    def circuits(self) -> List[Union[QuantumCircuit, Schedule]]:
-        """Return the circuits or pulse schedules for this job.
```

Comment:
We can still return the circuits. It just needs to be retrieved from the job parameters.